packages/@aws-cdk/aws-efs/test/efs-file-system.test.ts

```diff
@@ -240,3 +240,46 @@ test('can specify backup policy', () => {
     },
   }));
 });
+
+test('can create when using a VPC with multiple subnets per availability zone', () => {
+  // WHEN
+  new FileSystem(stack, 'EfsFileSystem', {
+    vpc,
+  });
+  // THEN
+  expectCDK(stack).to(haveResource('AWS::EFS::FileSystem', {
+    DeletionPolicy: 'Retain',
+    UpdateReplacePolicy: 'Retain',
+  }, ResourcePart.CompleteDefinition));
+  expectCDK(stack).to(countResourcesLike('AWS::EC2::Subnet', 2, {
+    AvailabilityZone: {
+      'Fn::Select': [
+        0,
+        {
+          'Fn::GetAZs': '',
+        },
+      ],
+    },
+  }));
+  expectCDK(stack).to(countResourcesLike('AWS::EC2::Subnet', 2, {
+    AvailabilityZone: {
+      'Fn::Select': [
+        1,
+        {
+          'Fn::GetAZs': '',
+        },
+      ],
+    },
+  }));
+  expectCDK(stack).to(countResources('AWS::EFS::MountTarget', 2));
+  expectCDK(stack).to(haveResource('AWS::EFS::MountTarget', {
+    SubnetId: {
```

Comment:
Why does this ensure that the mount targets are created in different az's? 

Seems to me that the subnets you are asserting on might be in the same AZ.

I think your previous approach was good, creating a vpc with a single az and two subnets, you just needs to assert only a single mount target was created even though there are two subnets in the VPC.